packages/common/src/TransactionUtils.js

```diff
@@ -0,0 +1,128 @@
+    const gasPriceScalingFunction = ynatm.DOUBLES;
+    const retryDelay = 60000;
+    const minGasPrice = transactionConfig.gasPrice;
+    const maxGasPrice = 2 * 3 * minGasPrice;
+
+    const receipt = await ynatm.send({
+      sendTransactionFunction: gasPrice => transaction.send({ ...transactionConfig, gasPrice }),
+      minGasPrice,
+      maxGasPrice,
+      gasPriceScalingFunction,
+      delay: retryDelay
+    });
+
+    return { receipt, returnValue, transactionConfig };
+  } catch (error) {
+    error.type = "send";
+    throw error;
+  }
+};
+
+/**
+ * Checks if an account has a pending transaction.
+ * @param {*Object} web3.js object for making queries and accessing Ethereum related methods.
+ * @param {*string} account account to check.
+ * @return Bool true if the account has pending transaction and false if no pending transaction.
+ */
+const accountHasPendingTransactions = async (web3, account) => {
+  const [currentMindedTransactions, currentTransactionsIncludingPending] = await Promise.all([
+    web3.eth.getTransactionCount(account, "latest"),
+    getPendingTransactionCount(web3, account)
+  ]);
+  return currentTransactionsIncludingPending > currentMindedTransactions;
+};
+
+/**
+ * Returns the number of pending transaction an account has. This method uses `provider.send` syntax. This undocumented
+ * web3 method lets you call direct jsonrpc methods on the provider. This method differs from the web3.js getTransactionCount
+ * by including mempool transactions. see https://infura.io/docs/ethereum/json-rpc/eth-getTransactionCount
+ * @param {*Object} web3.js object for making queries and accessing Ethereum related methods.
+ * @param {*string} account account to check.
+ * @returns number representing the number of transactions, including pending.
+ */
+const getPendingTransactionCount = async (web3, account) => {
+  const sendRpc = util.promisify(web3.currentProvider.send).bind(web3.currentProvider);
+
+  const rpcResponse = await sendRpc({
+    jsonrpc: "2.0",
+    method: "eth_getTransactionCount",
+    params: [account, "pending"],
+    id: Math.round(Math.random() * 100000)
+  });
+  return web3.utils.toDecimal(rpcResponse.result);
+};
+
+/**
+ * Blocking code until a specific block number is mined. Will re-fetch the current block number every 500ms. Useful when
+ * using methods called on contracts directly after state changes. Max blocking time should be ~ 15 seconds.
+ * @param {Object} web3 Provider from Truffle/node to connect to Ethereum network.
+ * @param {number} blockerBlockNumber block execution until this block number is mined.
+ */
+const blockUntilBlockMined = async (web3, blockerBlockNumber) => {
+  if (argv._.indexOf("test") !== -1) return;
+  for (;;) {
+    const currentBlockNumber = await web3.eth.getBlockNumber();
+    if (currentBlockNumber >= blockerBlockNumber) break;
+    await new Promise(r => setTimeout(r, 500));
```

Comment:
Seems potentially useful to parameterize the delay time between `setTimeout`'s, and setting it by default to 500.

Reply:
can do. this util was already in the repo but I agree setting a param as default to 500 would be better.